trustwallet/wallet-core
# Bitcoin Message Signing and Verification

## Description

Some Bitcoin wallets support a message signing & verification format, to create a proof (a signature) that someone has access to the private keys of a specific address.
This feature currently works on old legacy addresses only, and is not widely used, but we can add it nonetheless.

Note that C++ methods are added, not currently exposed over C interface.

Fixes #2597 .

## How to test

1. See unit tests in `MessageSignerTests`.
2. Generate some signature with the unit tests, and verify them with an online 3rd party verifier:
https://www.bitcoin.com/tools/verify-message/
https://www.verifybitcoinmessage.com/
https://reinproject.org/bitcoin-signature-tool/
3. Pick a private key, generate WIF format, generate corresponding legacy address.  Use one of the services above to create a signature. Validate with unit tests.
3. Using `bitcoin-cli`, generate a test wallet (non-descriptor based), get new address (legacy!), dump the private key, generate a signature. Validate that signature with unit tests.

## Types of changes





* New feature (non-breaking change which adds functionality)



## Checklist




If you're adding a new blockchain
